code/modules/mechs/mech_damage.dm

```diff
@@ -72,15 +92,39 @@
 	return def_zone //Careful with effects, mechs shouldn't be stunned
 
 /mob/living/exosuit/getarmor(def_zone, type)
-	if(body?.armor_plate)
-		var/body_armor = body.armor_plate?.armor.getRating(type)
-		if(body_armor) return body_armor
+	if(!def_zone)
+		def_zone = ran_zone()
+	var/obj/item/mech_component/hit_zone = zoneToComponent(def_zone)
+	var/armor = hit_zone?.armor.getRating(type)
+	if(armor)
+		return armor
 	return 0
 
 /mob/living/exosuit/updatehealth()
 	if(body) maxHealth = body.mech_health
 	health = maxHealth - (getFireLoss() + getBruteLoss())
 
+/mob/living/exosuit/damage_through_armor(damage, damagetype, def_zone, attack_flag, armor_divisor, used_weapon, sharp, edge, wounding_multiplier, list/dmg_types, return_continuation, dir_mult)
+	var/obj/item/mech_component/comp = zoneToComponent(def_zone)
+	var/armor_def = comp.armor.getRating(attack_flag)
+	var/deflect_chance = ((comp.shielding + armor_def)*0.5) - (armor_divisor*5)
+	if(prob(deflect_chance)) // Energy weapons have no physical presence, I would suggest adding a damage type check here later, not touching it for now because it affects game balance too much
```

Comment:
The way handheld shields handle them right now could work - apply deflection as a straight reduction to % of damage dealt by energy weapons. This reflects the energy not transferring to the target completely.

Reply:
I didn't type that, sorry